### PR TITLE
Test retry of failed initial db connection

### PR DESCRIPTION
**Pull request by c4ffein**
Tests for https://github.com/Scille/parsec-cloud/pull/1258#pullrequestreview-445963430

### CODE REVIEW

File: tests/backend/test_postgres_db.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+import pytest
+
+from parsec.backend.config import PostgreSQLBlockStoreConfig
+
+
+@pytest.mark.trio
+@pytest.mark.postgresql
+async def test_postgresql_connection_ok(
+    postgresql_url, alice, bob, backend_factory, backend_sock_factory
```
Comment (touilleMan) [inline tests/backend/test_postgres_db.py:13]: `alice`, `bob` & `backend_sock_factory` fixtures needed ? (same thing for the other tests' fixtures)

File: tests/backend/test_postgres_db.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+@pytest.mark.trio
+@pytest.mark.postgresql
+async def test_postgresql_connection_not_ok_retrying(
+    postgresql_url, alice, bob, backend_factory, backend_sock_factory, recwarn, caplog
+):
+    postgresql_url = re.sub(
+        ":(\\d+)/", lambda m: f":{str(int(m.group()[1:-1])+1)}/", postgresql_url
+    )
+    with pytest.raises(OSError):
```
Comment (touilleMan) [inline tests/backend/test_postgres_db.py:47]: `with pytest.raises(OSError) as exc:` to then test `str(exc.value) == "<error message about the connection>"` ?

File: tests/backend/test_postgres_db.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+from parsec.backend.config import PostgreSQLBlockStoreConfig
+
+
+@pytest.mark.trio
+@pytest.mark.postgresql
+async def test_postgresql_connection_ok(
+    postgresql_url, alice, bob, backend_factory, backend_sock_factory
+):
+    async with backend_factory(
+        config={"blockstore_config": PostgreSQLBlockStoreConfig(), "db_url": postgresql_url}
```
Comment (touilleMan) [inline tests/backend/test_postgres_db.py:16]: why `"blockstore_config": PostgreSQLBlockStoreConfig()` ? I would think `blockstore` fixtures returns `PostgreSQLBlockStoreConfig()` for postgresql tests (and this test is marked as postgresql only)

File: tests/backend/test_postgres_db.py [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        pass
+
+
+@pytest.mark.trio
+@pytest.mark.postgresql
+async def test_postgresql_connection_not_ok(
+    postgresql_url, alice, bob, backend_factory, backend_sock_factory, caplog
+):
+    postgresql_url = re.sub(
+        ":(\\d+)/", lambda m: f":{str(int(m.group()[1:-1])+1)}/", postgresql_url
```
Comment (touilleMan) [inline tests/backend/test_postgres_db.py:27]: This is too complex for it own good ^^

Besides we should use `unused_tcp_port` fixture to ensure the address is invalid
```python
bad_postgresql_url = f"postgresql://localhost:{unused_tcp_port}/dummy"
```

File: tests/backend/test_postgres_db.py [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    )
+    with pytest.raises(OSError):
+        async with backend_factory(
+            config={
+                "blockstore_config": PostgreSQLBlockStoreConfig(),
+                "db_url": postgresql_url,
+                "db_first_tries_number": 4,
+                "db_first_tries_sleep": 0,
```
Comment (touilleMan) [inline tests/backend/test_postgres_db.py:53]: we can add a non-null sleep time, then use `autojump_clock` fixture and enclose within a `trio.fail_after(4 * sleep_time + 1)` ;-)

File: tests/backend/test_postgres_db.py (hunk 1 above)
Comment (c4ffein) [reply tests/backend/test_postgres_db.py:13]: Nope

File: tests/backend/test_postgres_db.py (hunk 2 above)
Comment (c4ffein) [reply tests/backend/test_postgres_db.py:47]: Done

File: tests/backend/test_postgres_db.py (hunk 3 above)
Comment (c4ffein) [reply tests/backend/test_postgres_db.py:16]: Does work, but is not even needed actually

File: tests/backend/test_postgres_db.py (hunk 4 above)
Comment (c4ffein) [reply tests/backend/test_postgres_db.py:27]: Ofc...

File: tests/backend/test_postgres_db.py (hunk 5 above)
Comment (c4ffein) [reply tests/backend/test_postgres_db.py]: done

File: tests/backend/test_postgres_db.py [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+
+@pytest.mark.trio
+@pytest.mark.postgresql
+async def test_postgresql_connection_not_ok_retrying(
+    postgresql_url, backend_factory, caplog, unused_tcp_port, autojump_clock
+):
+    postgresql_url = f"postgresql://localhost:{unused_tcp_port}/dummy"
+    with pytest.raises(OSError):
+        with trio.fail_after(3 * 3 + 1):
```
Comment (touilleMan) [inline tests/backend/test_postgres_db.py:36]: image
:trollface:

File: tests/backend/test_postgres_db.py (hunk 2 above)
Comment (touilleMan) [reply tests/backend/test_postgres_db.py:35]: Did you forgot to push the changes ?

Comment (c4ffein) [reply tests/backend/test_postgres_db.py:35]: Nope, forgot one of the UT...

File: tests/backend/test_postgres_db.py (hunk 6 above)
Comment (c4ffein) [reply tests/backend/test_postgres_db.py:36]: :crying_cat_face:

**Pull request merged**